modules/tinymce/src/themes/silver/main/ts/ui/image/Images.ts

```diff
@@ -0,0 +1,84 @@
+import { AddEventsBehaviour, AlloyEvents, Behaviour, SimpleSpec } from '@ephox/alloy';
+import { Arr, Obj } from '@ephox/katamari';
+import { Css, Insert, Remove, SelectorFind, SugarElement } from '@ephox/sugar';
+import createDompurify from 'dompurify';
+
+export type ImageProvider = () => Record<string, string>;
+
+interface ImageSpec {
+  readonly tag: string;
+  readonly classes: string[];
+  readonly attributes?: Record<string, string>;
+  readonly behaviours?: Array<Behaviour.NamedConfiguredBehaviour<any, any, any>>;
+}
+
+const getInnerHTML = (url: string): string => createDompurify().sanitize(`
```

Comment:
This should be a component structure instead of raw HTML. So instead use stuff like:

```
components: [
  {
     tag: 'div',
     classes: [ 'blah' ]
  },
  components: [
     {
        dom: {
           tag: 'img',
           attributes: { src: imageUrl }
         }
     }
  ]
]
```

That is likely why it's complaining about injection and even with that you would need to URL encode.

- [ ] Switch to components instead of raw HTML

Reply:
done